### ISSUE TITLE
Readd checks for 'Multi-join relationship propagation' in the right scope

### ISSUE THREAD

**Issue by andrus** [nhl/link-rest#237]
Per #234 we lost some checks for a few specific conditions where we don't support multi-join relationships (in updates ... anywhere else?). We need to restore those checks and exceptions in the correct scope.

**Comment by atomashpolskiy** [comment]
I'm looking at this now, and it seems to me that it would be enough to check if the relationship is multi-join in `com.nhl.link.rest.runtime.parser.EntityJsonTraverser#processRelationship`. This will require either adding a new method `LrPersistentRelationship.isMultiJoin():boolean` or merging https://github.com/nhl/link-rest/pull/228, that introduces `LrPersistentRelationship.getJoins()` method.

**Comment by andrus** [comment]
Yeah, it was in `EntityJsonTraverser` initially. May not be the ideal place either, but we can restore it there. And maybe write a test for this condition, so that we don't miss it on future refactoring.
